wallet/src/main/java/com/android/identity_credential/wallet/presentation/PresentationFlowActivity.kt

```diff
@@ -0,0 +1,294 @@
+
+                        // update the KeyUnlockData
+                        keyUnlockData = unlockData
+
+                        val successfulBiometricResult = showBiometricPrompt(
+                            activity = this,
+                            title = resources.getString(R.string.presentation_biometric_prompt_title),
+                            subtitle = resources.getString(R.string.presentation_biometric_prompt_subtitle),
+                            cryptoObject = cryptoObject,
+                            userAuthenticationTypes = setOf(
+                                UserAuthenticationType.BIOMETRIC,
+                                UserAuthenticationType.LSKF
+                            ),
+                            requireConfirmation = false
+                        )
+
+                        // if the prompt was not successful (unable to authenticate), throw Exception
+                        if (!successfulBiometricResult) {
+                            throw PromptUnsuccessfulException("Biometric")
+                        }
+                    }
+
+                    // show Passphrase prompt
+                    is SoftwareSecureArea -> {
+                        val softwareKeyInfo =
+                            credential.secureArea.getKeyInfo(credential.alias) as SoftwareKeyInfo
+
+                        val passphrase = showPassphrasePrompt(
+                            activity = this,
+                            constraints = softwareKeyInfo.passphraseConstraints!!,
+                            checkWeakPassphrase = softwareKeyInfo.isPassphraseProtected,
+                        )
+
+                        if (passphrase.isEmpty()) {
+                            throw PromptUnsuccessfulException("Passphrase")
+                        }
+
+                        // use the passphrase that the user entered to create the KeyUnlockData
+                        keyUnlockData = SoftwareKeyUnlockData(passphrase)
+                    }
+
+                    // for secure areas not yet implemented
+                    else -> {
+                        throw IllegalStateException("No prompts implemented for secure area ${credential.secureArea}")
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Exception thrown when a prompt is cancelled during a Presentation, rather than throwing an
+     * [IllegalStateException]. This is to differentiate unsuccessful prompts from other events
+     * that throw [IllegalStateException] so an appropriate message can be shown if needed.
+     *
+     * @param promptType the prompt type that was unsuccessful (user cancelled or unauthorized)
+     */
+    class PromptUnsuccessfulException(promptType: String) :
+        Exception("The $promptType Prompt Not Successful!")
+
+    /**
+     * Helper class that provides Mdoc-related functions for [PresentationFlowAcivity].
+     */
+    private class MdocHelper {
```

Comment:
It looks like this helper class doesn't need to exist and all the methods in it could just be stand-alone methods/functions... or am I missing something?